Sources/ShopifyCheckoutSheetKit/CheckoutError.swift

```diff
@@ -41,3 +43,52 @@ public enum CheckoutError: Swift.Error {
 	/// In event of checkoutExpired, a new checkout url will need to be generated
 	case checkoutExpired(message: String)
 }
+
+internal enum CheckoutErrorGroup: String, Codable {
+	/// An authentication error
+	case authentication
+	/// A terminal checkout error which cannot be handled
+	case unrecoverable
+	/// A checkout-related error, such as failure to receive a receipt or progress through checkout
+	case checkout
+	/// The checkout session has expired and is no longer available
+	case expired
+	/// The error sent by checkout is not supported
+	case unsupported
+}
+
+internal struct CheckoutErrorEvent: Codable {
+	public let group: CheckoutErrorGroup
+	public let code: String?
```

Comment:
https://github.com/Shopify/checkout-native-sdk/blob/c3ce7921f97d2865983273a600683e304985af8a/ios/ShopifyCheckout/ShopifyCheckout/Models/Errors/InternalErrorPayload.swift#L30

Wondering if it's worth similar typing to make it a bit more understandable?

Reply:
Hmm from what I've seen we're not emitting error codes over the bridge. Do you mean we should map error strings to ints?